Signal/src/ViewControllers/ContactViewController.swift

```diff
@@ -0,0 +1,686 @@
+            }))
+            stackView.addArrangedSubview(createCircleActionButton(text: NSLocalizedString("ACTION_VIDEO_CALL",
+                                                                                          comment: "Label for 'video call' button in contact view."),
+                                                                  actionBlock: { [weak self] _ in
+                                                                    guard let strongSelf = self else { return }
+                                                                    strongSelf.didPressVideoCall()
+            }))
+            topView.addSubview(stackView)
+            stackView.autoPinEdge(.top, to: .bottom, of: lastView, withOffset: 20)
+            stackView.autoPinLeadingToSuperviewMargin(withInset: hMargin)
+            stackView.autoPinTrailingToSuperviewMargin(withInset: hMargin)
+            lastView = stackView
+        case .systemContactWithoutSignal:
+            // Show invite button for system contacts without a Signal account.
+            let inviteButton = createLargePillButton(text: NSLocalizedString("ACTION_INVITE",
+                                                                                          comment: "Label for 'invite' button in contact view."),
+                                                     actionBlock: { [weak self] _ in
+                                                        guard let strongSelf = self else { return }
+                                                        strongSelf.didPressInvite()
+            })
+            topView.addSubview(inviteButton)
+            inviteButton.autoPinEdge(.top, to: .bottom, of: lastView, withOffset: 20)
+            inviteButton.autoPinLeadingToSuperviewMargin(withInset: 55)
+            inviteButton.autoPinTrailingToSuperviewMargin(withInset: 55)
+            lastView = inviteButton
+        case .nonSystemContact:
+            // Show no action buttons for contacts not in user's device contacts.
+            break
+        case .noPhoneNumber:
+            // Show no action buttons for contacts without a phone number.
+            break
+        case .unknown:
+            let activityIndicator = UIActivityIndicatorView(activityIndicatorStyle: .whiteLarge)
+            topView.addSubview(activityIndicator)
+            activityIndicator.autoPinEdge(.top, to: .bottom, of: lastView, withOffset: 10)
+            activityIndicator.autoHCenterInSuperview()
+            lastView = activityIndicator
+            break
+        }
+
+        lastView.autoPinEdge(toSuperviewEdge: .bottom, withInset: 15)
+
+        let bottomView = UIView.container()
+        bottomView.backgroundColor = UIColor.white
+        bottomView.layoutMargins = .zero
+        bottomView.preservesSuperviewLayoutMargins = false
+        rootView.addSubview(bottomView)
+        bottomView.autoPinEdge(.top, to: .bottom, of: topView, withOffset: 0)
```

Comment:
No biggie, but can't this just be `bottomView.autoPinEdge(.top, to: .bottom, of: topView)`

Reply:
Removed.